File: MeTalk/Recorder/Views/MTRecordButton.m
//
//  MTRecordButton.m
//  MeTalk
//
//  Created by yu chung hyun on 2016. 3. 6..
//  Copyright © 2016년 yu chung hyun. All rights reserved.
//

#import "MTRecordButton.h"

@implementation MTRecordButton

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if(self)
    {
        [self initialize];
    }
    return self;
}

- (instancetype)initWithCoder:(NSCoder *)aDecoder
{
    self = [super initWithCoder:aDecoder];
    if(self)
    {
        [self initialize];
    }
    return self;
}

- (void)initialize
{
    self.layer.borderColor = [UIColor blackColor].CGColor;
    self.layer.backgroundColor = [UIColor redColor].CGColor;
    

    [self setImage:[UIImage imageNamed:@"audioBtnRecording"] forState:UIControlStateNormal];
    
    
    [self _resizeFrame];

}

- (void)_resizeFrame
{
    self.layer.cornerRadius = CGRectGetWidth(self.frame) / 2;
}

#pragma mark -- effect
- (void) setHighlighted: (BOOL) highlighted {
    [super setHighlighted: highlighted];
    self.alpha = highlighted ? 0.5f : 1.0f;
}

#pragma mark -- touch event handler
- (void)actionTouchDown:(id)sender
{
    NSLog(@"hi");
}
- (void)actionTouchUp:(id)sender
{
    NSLog(@"low");
}


@end

File: MeTalk/Recorder/Views/MTRecordButton.swift
//
//  MTRecordButton.swift
//  MeTalk
//
//  Created by yu chung hyun on 2016. 5. 11..
//  Copyright © 2016년 yu chung hyun. All rights reserved.
//

import UIKit

class MTRecordButton: UIButton {

    override var highlighted: Bool {
        didSet {
            self.alpha = (self.highlighted) ? 0.5 : 1.0
        }
    }
    
    override init(frame: CGRect) {
        
        super.init(frame: frame)
        
        self.initialize()

    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)

        self.initialize()
    }

    
    private func initialize() -> Void {
        self.layer.borderColor = UIColor.blackColor().CGColor;
        self.layer.backgroundColor = UIColor.colorRGB(0x44, 0x66, 0x77).CGColor;
        
        self.setImage(UIImage(named:"audioBtnRecording"), forState: .Normal)
        
        self.resizeFrame()
    }
    
    private func resizeFrame() -> Void {
        self.layer.cornerRadius = CGRectGetWidth(self.frame) / 2
    }
    
    
}
